src/finch/compiled.py

```diff
@@ -4,7 +4,7 @@
 from .tensor import Tensor
 
 
-def compiled(func):
+def compiled(func, opt="galley"):
```

Comment:
I'm not sure if this way you can do:
```python
@sparse.compiled(opt="galley")
def my_function(...):
    ...
```
WDYT?

I think one way is to have one function nesting more (like [here](https://stackoverflow.com/a/5929165/6717327)):
```python
def compiled(opt=""):
    def inner(func):
        @wraps(func)
        def wrapper_func(*args, **kwargs):
            new_args = []
            for arg in args:
                if isinstance(arg, Tensor) and not jl.isa(arg._obj, jl.Finch.LazyTensor):
                    new_args.append(Tensor(jl.Finch.LazyTensor(arg._obj)))
                else:
                    new_args.append(arg)
            result = func(*new_args, **kwargs)
            kwargs = {"ctx": get_scheduler(name=opt)} if opt != "" else {}
            result_tensor = Tensor(jl.Finch.compute(result._obj, **kwargs))
            return result_tensor
        return wrapper_func

    return inner
```

Reply:
Ahh, I had misunderstood the decorator model. This makes more sense. I'll switch it to this version.